models/release.go

```diff
@@ -369,6 +370,21 @@ func DeleteReleaseByID(id int64, doer *User, delTag bool) error {
 		go HookQueue.Add(rel.Repo.ID)
 	}
 
+	uuids := make([]string, 0, len(rel.Attachments))
+
+	for i := range rel.Attachments {
+		attachment := rel.Attachments[i]
+		if err := os.RemoveAll(attachment.LocalPath()); err != nil {
+			return err
+		}
+
+		uuids = append(uuids, attachment.UUID)
+	}
+
+	if _, err := x.In("uuid", uuids).Delete(new(Attachment)); err != nil {
```

Comment:
Why not just `x.Delete(&Attachment{ReleaseID: id})`? And this code should be before webhook.